packages/shared/src/hooks/useOrganizationList.tsx

```diff
@@ -1,39 +1,248 @@
+type UseOrganizationList = (params?: UseOrganizationListParams) => UseOrganizationListReturn;
+
+export const useOrganizationList: UseOrganizationList = params => {
+  const { userInvitations } = params || {};
+
+  const shouldUseDefaults = typeof userInvitations === 'boolean' && userInvitations;
+  const [paginatedPage, setPaginatedPage] = useState(shouldUseDefaults ? 1 : userInvitations?.initialPage ?? 1);
+
+  // Cache initialPage and initialPageSize until unmount
+  const initialPageRef = useRef(shouldUseDefaults ? 1 : userInvitations?.initialPage ?? 1);
+  const initialPageSizeRef = useRef(shouldUseDefaults ? 10 : userInvitations?.initialPageSize ?? 10);
+
+  const triggerInfinite = shouldUseDefaults ? false : !!userInvitations?.infinite;
+  const internalKeepPreviousData = shouldUseDefaults ? false : !!userInvitations?.keepPreviousData;
+  const internalStatus = shouldUseDefaults ? 'pending' : userInvitations?.status ?? 'pending';
 
-export const useOrganizationList: UseOrganizationList = () => {
   const clerk = useClerkInstanceContext();
   const user = useUserContext();
 
+  const paginatedParams =
+    typeof userInvitations === 'undefined'
+      ? undefined
+      : {
+          initialPage: paginatedPage,
+          initialPageSize: initialPageSizeRef.current,
+          status: internalStatus,
+        };
+
+  const canFetch = !!(clerk.loaded && user);
+
+  // Some gymnastics to adhere to the rules of hooks
+  // We need to make sure useSWR is called on every render
+  const fetchInvitations = !clerk.loaded
+    ? () => ({ data: [], total_count: 0 } as ClerkPaginatedResponse<UserOrganizationInvitationResource>)
+    : () => user?.getOrganizationInvitations(paginatedParams);
+
+  const {
+    data: userInvitationsData,
+    isValidating: userInvitationsValidating,
+    isLoading: userInvitationsLoading,
+    error: userInvitationsError,
+    mutate: userInvitationsMutate,
+  } = useSWR(
+    !triggerInfinite && canFetch && paginatedParams ? cacheKey('userInvitations', user, paginatedParams) : null,
+    fetchInvitations,
+    { keepPreviousData: internalKeepPreviousData },
+  );
+
+  const getInfiniteKey = (
+    pageIndex: number,
+    previousPageData: ClerkPaginatedResponse<UserOrganizationInvitationResource> | null,
+  ) => {
+    if (!canFetch || !paginatedParams || !triggerInfinite) {
+      return null;
+    }
+
+    return cacheKey('userInvitations', user, {
+      initialPage: initialPageRef.current + pageIndex,
+      initialPageSize: initialPageSizeRef.current,
+      status: internalStatus,
+    });
+  };
+
+  const {
+    data: userInvitationsDataInfinite,
+    isLoading: userInvitationsLoadingInfinite,
+    isValidating: userInvitationsInfiniteValidating,
+    error: userInvitationsInfiniteError,
+    size,
+    setSize,
+    mutate: userInvitationsInfiniteMutate,
+  } = useSWRInfinite(getInfiniteKey, ({ initialPage, initialPageSize, status }) => {
+    return !clerk.loaded || !user
```

Comment:
❓ I believe this hook should trigger the fetch only if triggerInfinite is set to true

Reply:
It does check `getInfiniteKey`. The triggering is decided based on the key not the fetcher function. I updated that part and removed the conditional from this place to avoid confusions